src/main/cpp/blaze.cc

```diff
@@ -1073,58 +1084,37 @@ static bool AreStartupOptionsDifferent(
     options_different = true;
   }
 
-  // Args in running_server_args that are not in requested_args.
-  bool found_missing_args = false;
+  std::unordered_set<string> missing_from_new;
+  vector<string> missing_from_old;
+
+  // Add all of 'running_server_args' to 'missing_from_new', then remove all of
```

Comment:
I have to say, I don't really get this reasoning, but it looks very elegant, is less code than before and I'm also tired, so maybe this is fine. 😅

Reply:
It's a roundabout way to compute the difference of two sets A and B. (Except the sets are given as vectors, although all elements should be unique, because they are the canonical set of startup flags which should not have duplicates.)

The logic adds all elements from `A` to a set, then removes all elements of `B` from the set. What remains in the set is `A \ B`. If an element could not be removed from the set, then it was not there, so I add those to a vector, which yields `B \ A`.

I'm adding a test to verify this.